EIPS/eip-7892.md

```diff
@@ -72,24 +84,75 @@ To facilitate these changes on the execution layer, the `blobSchedule` object sp
 }
 ```
 
-On the consensus layer, a new parameter is added to the configuration:
+### Consensus layer configuration
 
-```
-BLOB_SCHEDULE:
-  - EPOCH: 348618
+A new `BLOB_PARAMETER_SCHEDULE` field is added to consensus layer configuration, containing a sequence of entries representing blob parameter changes after `ELECTRA_FORK_EPOCH`. This replaces the deprecated `MAX_BLOB_PER_BLOCK{_FORKNAME}` convention for future forks, while preserving existing entries for previous forks.
+
+Entry rules:
+
+- One entry per fork that changes blob parameters (regular or BPO forks).
+- `EPOCH` and `MAX_BLOBS_PER_BLOCK` required for all entries.
+- `FORK_VERSION` required only for BPO forks (regular forks specify this in source).
+
+```yaml
+BLOB_PARAMETER_SCHEDULE:
+  - EPOCH: 400000     ## A future anonymous BPO fork carrying a fork version
+    FORK_VERSION: 0x09000000
     MAX_BLOBS_PER_BLOCK: 24
-  - EPOCH: 355368
+  - EPOCH: 420000     ## A future anonymous BPO fork carrying a fork version
+    FORK_VERSION: 0x0A000000
     MAX_BLOBS_PER_BLOCK: 56
+  - EPOCH: 440000     ## A future named fork introducing blob parameter changes
+    MAX_BLOBS_PER_BLOCK: 72
 ```
 
 The parameters and schedules above are purely illustrative. Actual values and schedules are beyond the scope of this specification.
 
-### Requirements
+**Requirements:**
+
+- Execution and consensus clients **MUST** share consistent BPO fork schedules.
+- The timestamp in EL's `blobSchedule` **MUST** align with the start of the epoch specified in the consensus layer configuration.
+- The `max` field in `blobSchedule` **MUST** equal the `MAX_BLOBS_PER_BLOCK` value in the consensus layer configuration.
+
+### Modified `compute_fork_version`
+
+The `compute_fork_version` helper is updated to account for BPO forks:
+
+```python
+def compute_fork_version(epoch: Epoch, blob_schedule: Sequence[BlobScheduleEntry]) -> Version:
+  # Start with named forks.
+  forks = [
+      (ELECTRA_FORK_EPOCH, ELECTRA_FORK_VERSION),
+      (DENEB_FORK_EPOCH, DENEB_FORK_VERSION),
+      (CAPELLA_FORK_EPOCH, CAPELLA_FORK_VERSION),
+      (BELLATRIX_FORK_EPOCH, BELLATRIX_FORK_VERSION),
+      (ALTAIR_FORK_EPOCH, ALTAIR_FORK_VERSION),
+  ]
+  
+  # Add blob schedule entries that define fork versions (therefore representing BPO forks).
+  bpo_forks = [
+      (entry.epoch, entry.fork_version) 
+      for entry in blob_schedule 
+      if entry.fork_version is not None
+  ]
+  forks.extend(bpo_forks)
+  
+  forks.sort(reverse=True)
```

Comment:
Let's order `forks` chronologically, so Altair is first and Electra is last.

Currently, this results in: `bpo3`, `bpo2`, `bpo1`, `altair`, `bellatrix`, ..., `electra` which is wrong.

It should be `bpo3`, `bpo2`, `bpo1`, `electra`, `deneb`, ..., `altair` right?

Reply:
Also, this method is not going to work properly in the future when we add more regular forks. If we append a value to `forks` then, for example, it will be: `bpo3`, `bpo2`, `bpo1`, `gloas`, `fulu`, `electra`, `deneb`, ..., `altair` which is wrong. Because `gloas` is probably newer than `bpo3`.

And this reminds me. Let's add `fulu` to the list here.